pkg/operation/botanist/controlplane.go

```diff
@@ -1089,39 +1111,67 @@ func (b *Botanist) DeployETCD(ctx context.Context) error {
 		}
 
 		foundEtcd := true
-		statefulset := &appsv1.StatefulSet{}
-		if err := b.K8sSeedClient.Client().Get(ctx, kutil.Key(b.Shoot.SeedNamespace, fmt.Sprintf("etcd-%s", role)), statefulset); err != nil && !apierrors.IsNotFound(err) {
+		etcd := &druidv1alpha1.Etcd{}
+		if err := b.K8sSeedClient.Client().Get(ctx, kutil.Key(b.Shoot.SeedNamespace, fmt.Sprintf("etcd-%s", role)), etcd); client.IgnoreNotFound(err) != nil {
 			return err
 		} else if apierrors.IsNotFound(err) {
 			foundEtcd = false
 		}
 
-		if foundEtcd && hvpaEnabled {
+		name := fmt.Sprintf("etcd-%s", role)
+		if foundEtcd && etcd.Status.Etcd.Name != "" {
+			name = etcd.Status.Etcd.Name
+		}
+
+		foundStatefulset := true
+		statefulset := &appsv1.StatefulSet{}
+		if err := b.K8sSeedClient.Client().Get(ctx, kutil.Key(b.Shoot.SeedNamespace, name), statefulset); client.IgnoreNotFound(err) != nil {
+			return err
+		} else if apierrors.IsNotFound(err) {
+			foundStatefulset = false
+		}
+
+		// TODO(georgekuruvillak): Remove this, once HVPA support updating resources in CRD spec
+		if foundStatefulset && hvpaEnabled {
 			// etcd is already created AND is controlled by HVPA
 			// Keep the "resources" as it is.
 			for k := range statefulset.Spec.Template.Spec.Containers {
 				v := &statefulset.Spec.Template.Spec.Containers[k]
 				if v.Name == "etcd" {
-					etcd["etcdResources"] = v.Resources.DeepCopy()
+					values["etcd"].(map[string]interface{})["resources"] = v.Resources.DeepCopy()
+					break
+				} else if v.Name == "backup-restore" {
+					values["sidecar"].(map[string]interface{})["resources"] = v.Resources.DeepCopy()
 					break
 				}
 			}
 		}
 
 		if b.Shoot.HibernationEnabled {
-			// capture the old statefulset state
-			statefulset := &appsv1.StatefulSet{}
-			if err := client.IgnoreNotFound(b.K8sSeedClient.Client().Get(ctx, kutil.Key(b.Shoot.SeedNamespace, fmt.Sprintf("etcd-%s", role)), statefulset)); err != nil {
-				return err
-			}
-
 			// Restore the replica count from capture statefulset state.
-			etcd["replicas"] = 0
-			if statefulset.Spec.Replicas != nil {
-				etcd["replicas"] = *statefulset.Spec.Replicas
+			values["replicas"] = 0
+			if foundEtcd {
+				values["replicas"] = etcd.Spec.Replicas
+			} else {
+				// In case etcd resource is not deployed but statefulset exists.
+				if foundStatefulset && statefulset.Spec.Replicas != nil {
```

Comment:
You could combine this if statement with the else above, rigth?

Reply:
done